cmd/run.go

```diff
@@ -115,13 +116,19 @@ func NewRunCommand(stopCh <-chan struct{}) *cobra.Command {
 			p := proxy.New(restConfig, reqAuther, tokenReviewer,
 				secureServingInfo)
 
+			// try initializing provider
+			ctx := context.Background()
+			err = util.InitProviderUntil(ctx, oidcOptions, stopCh)
+			if err != nil {
+				return err
+			}
+
```

Comment:
I am not really a fan of this approach. We basically initialise the provider twice and we have no guarantees that the second time is actually finished after the first try.

The reason for this async setup is that Kubernetes wants to support the usecase where the OIDC server is actually running on the same cluster. Why do we need this to initialised before we start listening?

I am sure it would be fine to error with the error we get from AuthenticateToken.

Reply:
Thanks for the review!

> We basically initialise the provider twice and we have no guarantees that the second time is actually finished after the first try.

I am not a fan of this approach, either. I identified the above as one of the downsides in the PR description. I implemented an alternative in #93, but it requires a (very small) change to upstream (k8s.io/apimachinery) code.

> Why do we need this to initialised before we start listening?

Please let me know if the motivation I wrote in #92 was not clear, and I'll try to improve it.